src/components/Ui/Select.vue

```diff
@@ -1,22 +1,24 @@
-<script setup lang="ts">
-type Item = {
-  key: string;
+<script setup lang="ts" generic="T extends string | number, U extends readonly Item<T>[]">
```

Comment:
This seems hard to read in a single line, how about this:

```
type ItemKeyType = string | number;
type ItemArray<T extends ItemKeyType> = readonly Item<T>[];

<script setup lang="ts" generic="T extends ItemKeyType, U extends ItemArray<T>">
```

Reply:
I tried this and to me it seems harder to read, because we need to put those helper types below first line so to read you now need to jump back and forth multiple times to understand. In the end there is not much change in the first line, but we add two extra types.

Reply:
Ok I see what you mean.